### PR TITLE
[Kubernetes] source user information from user headers laid by oauth-proxy

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

- Enhancement / new feature

### Description

On Kubernetes, source user information from user headers laid by oauth-proxy and use this information to populate the upper right of the EnMasse Console.

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [x] Update CHANGELOG.md

### CODE REVIEW

File: console/console-init/oauth-proxy/cfg/oauth-proxy-kubernetes.cfg [hunk 1]
```diff
@@ -33,3 +33,5 @@ cookie_secret = "${SSO_COOKIE_SECRET}"
 cookie_domain = "${SSO_COOKIE_DOMAIN}"
 
 skip_provider_button = true
+
+pass_user_headers = true
```
Comment (lulf) [inline console/console-init/oauth-proxy/cfg/oauth-proxy-kubernetes.cfg]: With this being default, we can remove the extra argument added if impersonation is enabled: https://github.com/EnMasseProject/enmasse/blob/master/pkg/controller/consoleservice/consoleservice_controller.go#L903

Comment (k-wall) [reply console/console-init/oauth-proxy/cfg/oauth-proxy-kubernetes.cfg]: Done.

**Comment by k-wall** [comment]
@enmasse-ci run tests testcase=*ConsoleTest

**Comment by k-wall** [comment]
@enmasse-ci run tests testcase=*ConsoleTest

**Comment by k-wall** [comment]
#4237 was the root cause the testCreateAddrSpaceWithCustomAuthService failure, so merging.

**Pull request merged**